verl/single_controller/base/worker.py

```diff
@@ -138,14 +138,43 @@ def __init__(self, cuda_visible_devices=None) -> None:
         import os
 
         import torch
-        from packaging import version
 
-        ###
-        # [SUPPORT AMD: torch]
-        if torch.cuda.is_available() and "AMD" in torch.cuda.get_device_name() and version.parse(ray.__version__) < version.parse("2.45.0"):
-            os.environ["CUDA_VISIBLE_DEVICES"] = os.environ.get("ROCR_VISIBLE_DEVICES")
-            os.environ["LOCAL_RANK"] = os.environ.get("RAY_LOCAL_RANK")
-        ###
+        from verl.utils.ray_utils import ray_noset_visible_devices
+
+        is_ray_noset_visible_devices = ray_noset_visible_devices()
+
+        # Prevent use of clashing `{CUDA/HIP/ROCR}_VISIBLE_DEVICES``
+        rocr_val = os.environ.get("ROCR_VISIBLE_DEVICES", None)
```

Comment:
Ray >=2.45 will throw an error if ROCR_VISIBLE_DEVICES is used. In general, it can cause unexpected behavior in pytorch distributed.

Reply:
This is just for compatibility for ray < 2.45, and I don't think it will cause issues for Ray >=2.45, as ray will do that check if `ROCR_VISIBLE_DEVICES` is used from the beginning, and we take its values to `CUDA_VISIBLE_DEVICES` and unset it.